src/main/java/com/bankbazaar/test/service/EmployeeService.java

```diff
@@ -0,0 +1,117 @@
+package com.bankbazaar.test.service;
+
+
+import com.bankbazaar.test.entity.Department;
+import com.bankbazaar.test.entity.Employee;
+import com.bankbazaar.test.repository.EmployeeRepository;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+public class EmployeeService {
+
+    @Autowired
+    private EmployeeRepository employeeRepository;
+    private EmployeeRepository departmentRepository;
+    /**
+     * Insert to Employee table
+     * @param data
+     */
+    public Employee saveEmployee(Employee data)
+    {
+        try {
+            /**
+             * Get department data
+             */
+            final String uriGet = "http://localhost:8080/departmentById/"+data.getDepartmentId().getId();
+            HttpHeaders headers = new HttpHeaders();
+            headers.add("Authorization", "Basic YWRtaW46YWRtaW4xMjM0");
+            HttpEntity request = new HttpEntity(headers);
+
+            ResponseEntity<String> response = new RestTemplate().exchange(uriGet, HttpMethod.GET, request, String.class);
+
+            JsonObject departmentData = JsonParser.parseString(response.getBody()).getAsJsonObject();
```

Comment:
Is this for testing?

Reply:
Yes, i was trying to work with external apis.